AWSPollyTests/AWSPollyTests.m

```diff
@@ -374,4 +374,201 @@ - (void)testSynthesisOperations{
+        XCTAssertNil(task.error, @"Unexpected error in getSpeechSynthesisTask (testCaseLine %lu)", testCaseLine);
+        XCTAssertNotNil(task.result, @"Result unexpectedly nil in getSpeechSynthesisTask (testCaseLine %lu)", testCaseLine);
+        return nil;
+    }] waitUntilFinished];
+
+}
+
+// Creates a presigned URL request and downloads the file to local storage. NOTE: This does not play the
+// actual audio file, but leaves it on local storage for manual playback after the test.
+- (void)executeAndAssertURLBuilderForTestCase:(NSDictionary *)testCase {
+    AWSPollyVoiceId voiceId = [testCase[@"voiceId"] integerValue];
+    AWSPollyLanguageCode languageCode = [testCase[@"languageCode"] integerValue];
+    NSString *text = testCase[@"text"];
+    unsigned long testCaseLine = [testCase[@"testCaseLine"] integerValue];
+
+    AWSPollySynthesizeSpeechURLBuilderRequest *request = [[AWSPollySynthesizeSpeechURLBuilderRequest alloc] init];
+    [request setText:text];
+    [request setOutputFormat:AWSPollyOutputFormatMp3];
+    [request setVoiceId:voiceId];
+    [request setLanguageCode:languageCode];
+    [request setExpires:[NSDate dateWithTimeIntervalSinceNow:10*60]];
+
+    AWSPollySynthesizeSpeechURLBuilder *builder = [AWSPollySynthesizeSpeechURLBuilder defaultPollySynthesizeSpeechURLBuilder];
+
+    [[[[builder getPreSignedURL:request] continueWithBlock:^id _Nullable(AWSTask<NSURL *> * _Nonnull task) {
+        XCTAssertNil(task.error, @"Unexpected error in getPreSignedURL (testCaseLine %lu)", testCaseLine);
+        XCTAssertNotNil(task.result, @"Result unexpectedly nil in getPreSignedURL (testCaseLine %lu)", testCaseLine);
+        return task;
+    }] continueWithBlock:^id _Nullable(AWSTask * _Nonnull task) {
+        NSMutableURLRequest *request = [NSMutableURLRequest requestWithURL:task.result];
+        request.cachePolicy = NSURLRequestReloadIgnoringLocalCacheData;
+
+        dispatch_semaphore_t semaphore = dispatch_semaphore_create(0);
+        [[[NSURLSession sharedSession] dataTaskWithRequest:request completionHandler:^(NSData * _Nullable data, NSURLResponse * _Nullable response, NSError * _Nullable error) {
+            XCTAssertNil(error, @"Unexpected error in URL download (testCaseLine %lu), presigned URL is: %@", testCaseLine, ((NSURL *)task.result).absoluteString);
+
+            if (((NSHTTPURLResponse *)response).statusCode < 200 || ((NSHTTPURLResponse *)response).statusCode >= 300) {
+                XCTFail(@"Response status code is %ld (testCaseLine %lu)", (long)((NSHTTPURLResponse *)response).statusCode, testCaseLine);
+            }
+
+            NSString *fileName = [NSString stringWithFormat:@"polly-testURLBuilder-%lu.mp3", voiceId];
+            NSString *filePath = [[NSSearchPathForDirectoriesInDomains(NSDocumentDirectory, NSUserDomainMask, YES) firstObject] stringByAppendingPathComponent:fileName];
+
+            [data writeToFile:filePath atomically:YES];
```

Comment:
Should we test whether the file has data after writing? Otherwise this is unnecessary in the test.

Reply:
Great callout. Fixed.